File: PYTHON/Agregação.py
class Empregado:
    def __init__(self, nome, cargo, salario):
        self.nome = nome
        self.cargo = cargo
        self.salario = salario

    def detalhes(self):
        return f'Nome: {self.nome}, Cargo: {self.cargo}, Salário: R${self.salario}'

class Empresa:
    def __init__(self, nome):
        self.nome = nome
        self.empregados = []

    def adicionar_empregado(self, empregado):
        self.empregados.append(empregado)
    
    def listar_empregados(self):
        for empregado in self.empregados:
            print(empregado.detalhes())


empregado1 = Empregado('Pedro Souza', 'Desenvolvedor', 5000)
empregado2 = Empregado('Maria Souza', 'Analista', 4500)
empregado3 = Empregado('Carlos Pereira', 'Gerente', 6000)

empresa = Empresa('Tech Solutions')
empresa.adicionar_empregado(empregado1)
empresa.adicionar_empregado(empregado2)
empresa.adicionar_empregado(empregado3)

empresa.listar_empregados()

File: JAVA/Agregação.java
import java.util.ArrayList;
import java.util.List;

class Empregado {
    private String nome;
    private String cargo;
    private double salario;

    public Empregado(String nome, String cargo, double salario) {
        this.nome = nome;
        this.cargo = cargo;
        this.salario = salario;
    }

    public String getNome() {
        return nome;
    }

    public String getCargo() {
        return cargo;
    }

    public double getSalario() {
        return salario;
    }

    public String detalhes() {
        return "Nome: " + nome + ", Cargo: " + cargo + ", Salário: R$" + salario;
    }
}

class Empresa {
    private String nome;
    private List<Empregado> empregados;

    public Empresa(String nome) {
        this.nome = nome;
        this.empregados = new ArrayList<>();
    }

    public void adicionarEmpregado(Empregado empregado) {
        empregados.add(empregado);
    }

    public void listarEmpregados() {
        for (Empregado empregado : empregados) {
            System.out.println(empregado.detalhes());
        }
    }

    public static void main(String[] args) {
        Empregado empregado1 = new Empregado("Pedro Souza", "Desenvolvedor", 5000.00);
        Empregado empregado2 = new Empregado("Maria Souza", "Analista", 4500.00);
        Empregado empregado3 = new Empregado("Carlos Pereira", "Gerente", 6000.00);

        Empresa empresa = new Empresa("Tech Solutions");
        empresa.adicionarEmpregado(empregado1);
        empresa.adicionarEmpregado(empregado2);
        empresa.adicionarEmpregado(empregado3);

        System.out.println("Empregados da empresa " + empresa.nome + ":");
        empresa.listarEmpregados();
    }
}
